Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/Spire/QtPromise.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+std::tuple<boost::signals2::connection,
+    QtPromise<std::vector<Nexus::OrderImbalance>>>
+    LocalOrderImbalanceIndicatorModel::subscribe(
+    const boost::posix_time::ptime& start,
+    const boost::posix_time::ptime& end,
+    const OrderImbalanceSignal::slot_type& slot) {
+  m_signals.emplace_back(OrderImbalanceSignalConnection{
+      OrderImbalanceSignal(), start, end});
```

Comment:
Rename ```OrderImbalanceSignalConnection``` to ```Susbcription```, give it a constructor that takes a ```start``` and ```end```. That avoids the unneeded ```OrderImbalanceSignal()```.